File: 0150 evaluate-reverse-polish-notation/solution.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer> st = new Stack<>();
        int st1, st2;
        for (String token : tokens) {
            switch (token) {
                case "+":
                    st.push(st.pop() + st.pop());
                    break;
                case "-":
                    st2 = st.pop();
                    st1 = st.pop();
                    st.push(st1 - st2);
                    break;
                case "*":
                    st.push(st.pop() * st.pop());
                    break;
                case "/":
                    st2 = st.pop();
                    st1 = st.pop();
                    st.push(st1 / st2);
                    break;
                default:
                    st.push(Integer.parseInt(token));
            }
        }
        return st.pop();
    }
}

File: 0150 evaluate-reverse-polish-notation/solution.cs
public class Solution {
    public int EvalRPN(string[] tokens) {
        Stack<int> st = new();
        int st1, st2;
        foreach(var t in tokens) {
            switch(t){
                case "+" : st.Push(st.Pop()+st.Pop());
                break;
                case "-":  st2=st.Pop();
                            st1=st.Pop();
                            st.Push(st1-st2);
                break;
                case "*" : st.Push(st.Pop()*st.Pop());
                break;
                case "/" : st2=st.Pop();
                            st1=st.Pop();
                            st.Push(st1/st2);
                break;
                default: st.Push(int.Parse(t));
                break;
            }
        }
        return st.Pop();
    }
}
